src/Common/Abstracts/Custom_List_Table.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * The Abstract Custom List Table.
+ *
+ * @since TBD
+ *
+ * @package TEC/Common/Abstracts
+ */
+
+namespace TEC\Common\Abstracts;
+
+use TEC\Common\Admin\Abstract_Custom_List_Table as Base_Abstract;
+use TEC\Common\StellarWP\SchemaModels\Contracts\SchemaModel as Model;
+use TEC\Common\Contracts\Custom_Table_Repository_Interface as Repository;
+
+/**
+ * Class Abstract_Custom_List_Table
+ *
+ * @since TBD
+ *
+ * @package TEC/Common/Abstracts
+ */
+abstract class Custom_List_Table extends Base_Abstract {
+	/**
+	 * Returns the total number of items.
+	 *
+	 * @since TBD
+	 *
+	 * @return int The total number of items.
+	 */
+	protected function get_total_items(): int {
+		$repo = $this->get_repository();
+
+		$repo->set_use_default_args( false );
+		$result = $repo->by_args( $this->get_arguments() )->found();
+		$repo->set_use_default_args( true );
+
+		return $result;
+	}
+
+	/**
+	 * Returns the repository.
+	 *
+	 * @since TBD
+	 *
+	 * @return Repository The repository.
+	 */
+	abstract protected function get_repository(): Repository;
+
+	/**
+	 * Returns the items for the current page.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $per_page The number of items to display per page.
+	 *
+	 * @return array The items for the current page.
+	 */
+	protected function get_items( int $per_page ): array {
+		$repo = $this->get_repository();
+
+		$repo->set_use_default_args( false );
+		$results = $repo->by_args( $this->get_arguments() )->page( $this->get_pagenum() )->per_page( $per_page )->all();
+		$repo->set_use_default_args( true );
+
+		return $results;
+	}
+
+	/**
+	 * Returns whether the list is completely empty.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_empty(): bool {
+		$repo = $this->get_repository();
+		$repo->set_use_default_args( false );
+		$result = $repo->found();
+		$repo->set_use_default_args( true );
+
+		return 0 === $result;
+	}
+
+	/**
+	 * Returns the arguments to query the items.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	private function get_arguments(): array {
+		return array_merge( $this->get_args(), $this->get_instance_args() );
+	}
+
+	/**
+	 * Returns the instance arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	abstract protected function get_instance_args(): array;
```

Comment:
"Instance arguments" seems a little vague. Could more explanation be added to clarify exactly what that means? Or perhaps a different name?

Reply:
Perhaps something like `get_object_query_args()`? Since it's used as part of the querying process, that seems to make the most sense to me.

Will all child classes definitely need to have a custom array of arguments? If not, I would recommend changing this method to be a normal `protected` method that returns an empty array. Child classes that need to override can still do so.